tests/lms/config/__init___test.py

```diff
@@ -0,0 +1,193 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+import mock
+import pytest
+import pyramid.config
+
+from lms.config import configure
+from lms.config import SettingError
+
+
+@pytest.mark.usefixtures(
+    "env_setting",
+    "groupfinder",
+    "ACLAuthorizationPolicy",
+    "AuthTktAuthenticationPolicy",
+)
+class TestConfigure:
+    def test_it_returns_a_Configurator_with_the_deployment_settings_set(self, env_setting):
+        configurator = configure({})
+
+        assert isinstance(configurator, pyramid.config.Configurator)
+        # Just pick some settings at random to test.
+        for setting in ("jwt_secret", "google_client_id", "lms_secret"):
+            assert configurator.registry.settings[setting] == env_setting.return_value
+
+    def test_when_an_ENV_VAR_isnt_set_it_puts_None_into_the_settings(self, env_setting):
+        def side_effect(envvar_name, *args, **kwargs):  # pylint: disable=unused-argument
+            if envvar_name == "USERNAME":
+                return None
+            return mock.DEFAULT
+
+        env_setting.side_effect = side_effect
+
+        configurator = configure({})
+
+        assert configurator.registry.settings["username"] is None
+
+    def test_it_raises_if_a_required_environment_variable_is_missing(self, env_setting):
+        env_setting.side_effect = SettingError("error message")
+
+        with pytest.raises(SettingError, match="error message"):
+            configure({})
+
+    def test_the_aes_secret_setting_is_the_LMS_SECRET_env_var_as_a_byte_string(self, env_setting):
+        def side_effect(envvar_name, *args, **kwargs):  # pylint: disable=unused-argument
+            if envvar_name == "LMS_SECRET":
+                return "test_lms_secret"
+            return mock.DEFAULT
+
+        env_setting.side_effect = side_effect
+
+        configurator = configure({})
+
+        assert configurator.registry.settings["aes_secret"] == b"test_lms_secret"
```

Comment:
I am missing the magic of how this turns into a bytestring! But that's my own ignorance, I think?

Reply:
The `encode()` on this line turns it into an ASCII byte string: https://github.com/hypothesis/lms/blob/df063702e2e9851e97b1a5ce94d1bd8c8de8debf/lms/config/__init__.py#L40